From 7bcc15b25674b7443a69b887e2b223366a3ad26a Mon Sep 17 00:00:00 2001
From: Randy Dunlap <rdunlap@infradead.org>
Date: Thu, 23 Jul 2026 12:01:51 -0700
Subject: [PATCH 10176/15333] docs: xforms_lists: support DEFINE_IDTENTRY_IRQ()

Avoid kernel-doc warnings by supported DEFINE_IDTENTRY_IRQ() as a
function transform.

Warning: arch/x86/kernel/apic/apic.c:2157 function parameter
 'spurious_interrupt' not described in 'DEFINE_IDTENTRY_IRQ'
Warning: arch/x86/kernel/apic/apic.c:2157 expecting prototype for
 spurious_interrupt(). Prototype was for DEFINE_IDTENTRY_IRQ() instead

Signed-off-by: Randy Dunlap <rdunlap@infradead.org>
Signed-off-by: Jonathan Corbet <corbet@lwn.net>
Message-ID: <20260723190151.507295-1-rdunlap@infradead.org>
---
 tools/lib/python/kdoc/xforms_lists.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/lib/python/kdoc/xforms_lists.py b/tools/lib/python/kdoc/xforms_lists.py
index 4251f7c6673a..e3dda2fe8a53 100644
--- a/tools/lib/python/kdoc/xforms_lists.py
+++ b/tools/lib/python/kdoc/xforms_lists.py
@@ -90,6 +90,7 @@ class CTransforms:
         (CMatch("__(?:re)?alloc_size"), ""),
         (CMatch("__diagnose_as"), ""),
         (CMatch("DECL_BUCKET_PARAMS"), r"\1, \2"),
+        (CMatch("DEFINE_IDTENTRY_IRQ"), r"static void \1(struct pt_regs *regs, u32 vector)"),
         (CMatch("__cond_acquires"), ""),
         (CMatch("__cond_releases"), ""),
         (CMatch("__acquires"), ""),
-- 
2.55.0

